### ISSUE TITLE
Array validation does not seem to be working as expected.

### ISSUE THREAD

**Issue by godjen99** [laravel/framework#13212]
When trying the following code, I was expecting a false when checking if it passed, however that wasn't the case. All passed.

```
$data = [
    'companies' => ['dddd']
];

$validator = \Validator::make($data, [
    'companies.*.name' => 'required|max:255',
]);

dd($validator->passes());
```

Digging into the code, In the Illuminate/Validation/Validator.php on line 306, it's looking for "companies.*.name" in "companies.0", but doesn't find it (returns false), then looks to see if the regular expression "companies\.[^\.]+\.name" matches "companies.0", and it doesn't as well, so the rules don't get merged into the ruleset and allow for passing.

Is this intended? I'm not sure the fix, but I've been thinking on how to approach it, but don't want to try doing my first pull request on something thats intended ;)

Thanks for any help!

**Comment by pallmallshow** [comment]
Hi,

I think your data array must be like : 

```php
$data = [
    'companies' => [
        ['name' => 'dddd']
    ]
];
```

**Comment by godjen99** [comment]
Hello pallmallshow,

You're correct, that ends up failing, as expected. However my intent for the API endpoint I was creating was to force the need for a companies key and many companies (to create many companies at once. 

My mind read the 'companies.*.name' as "Require a companies key and any sub keys must have the"name" field". After playing around, I've found a work around, but doesn't seem to fit who I would have expected it to work. So the way I fixed it was like this:

```
$data = [
    'companies' => ['dddd']
];

$validator = \Validator::make($data, [
    'companies' => 'required|array',
    'companies.*' => 'required|array',
    'companies.*.name' => 'required|max:255',
]);

dd($validator->passes()); // now comes back as false, as expected.
```

I do wonder the point of the validation if I have to explicitly add the other rules. Is there any reason the validation shouldn't work like I thought it should? Which would be the fact I have a "required" on an array subset with that subset missing should result in failed validation.
